packages/core/src/application.ts

```diff
@@ -1,6 +1,6 @@
 import { Metadata } from './metadata';
 import { parseVersion, Version } from './version';
-
+import { type PropsBeforeEvaled } from '@sunmao-ui/shared';
```

Comment:
This type had better to be in `core` rather than `shared`. Because if `shared` imported a type( which is very likely), there will be cycle dependency.